internal-api/src/main/java/datadog/trace/util/PidHelper.java

```diff
@@ -79,11 +99,7 @@ public static Set<String> getJavaPids() {
     // Some JDKs don't have jvmstat available as a module, attempt to read from the hsperfdata
     // directory instead
     try (Stream<Path> stream =
-        Files.list(
-            Paths.get(
-                System.getProperty("java.io.tmpdir")
-                    + "hsperfdata_"
-                    + System.getProperty("user.name")))) {
+        Files.list(Paths.get(getOSTempDir() + "/hsperfdata_" + System.getProperty("user.name")))) {
```

Comment:
```suggestion
       // Emulating the hotspot way to enumerate the JVM processes using the perfdata file
       // https://github.com/openjdk/jdk/blob/d7cb933b89839b692f5562aeeb92076cd25a99f6/src/hotspot/share/runtime/perfMemory.cpp#L244
        Files.list(Paths.get(getOSTempDir() + "/hsperfdata_" + System.getProperty("user.name")))) {
```

Also, it would be good to take a look at the `PerfDataSaveFile` JVM arg and check the allowed formatting there. Although it is rather unlikely someone would change that we should at least have it recorded in the comments here for the future maintainers.

Reply:
**suggestion**: While taking a look it seems these can be turned off as well `-XX:-UsePerfData`.